cpp-ch/local-engine/Storages/SubstraitSource/SubstraitFileSource.cpp

```diff
@@ -65,15 +65,15 @@ static DB::Block getRealHeader(const DB::Block & header)
 }
 
 SubstraitFileSource::SubstraitFileSource(
-    DB::ContextPtr context_, const DB::Block & header_, const substrait::ReadRel::LocalFiles & file_infos)
+    DB::ContextPtr context_, const DB::Block & header_, const std::shared_ptr<substrait::ReadRel::LocalFiles> & file_infos)
```

Comment:
It seems no special reason to change file_info into a shared ptr if there is no need to hold it in `SubstraitFileSource`